modules/photons_control/clean.py

```diff
@@ -0,0 +1,188 @@
+from photons_control.script import FromGenerator
+
+from photons_app.tasks import task_register as task
+from photons_app.errors import PhotonsAppError
+
+from photons_messages import LightMessages, DeviceMessages
+
+
+def CleanConfig(indication=None, duration=None, **kwargs):
+    """
+    Returns a valid message that will either get or set the default clean cycle
+    configuration for the device.
+
+    For example:
+
+    .. code-block:: python
+
+        await target.send(CleanConfig(indication=True), ["d073d5000001", "d073d5000001"])
+    """
+
+    async def gen(reference, sender, **kwargs):
+        has_hev = DeviceMessages.GetVersion()
```

Comment:
This is what the capabilities system is for :)

So a better way here is using the capabilities plan like how SetZonesEffect works https://github.com/delfick/photons/blob/main/modules/photons_control/multizone.py#L272

You want the [has_hev](https://github.com/delfick/photons/blob/main/modules/photons_products/lifx.py#L71) option

(it's overridden by the products that have it, for example [here](https://github.com/delfick/photons/blob/main/modules/photons_products/registry.py#L555))

Reply:
I knew something better would exist. :)

Reply:
the gatherer system exists because getting version and host firmware and then combining the two everywhere was really annoying me a lot hahhah

Reply:
Weirdly, I used the gatherer for the `get_clean_default_config` task.